javascript/javascript3/week2/currencyCalculator.js

```diff
@@ -0,0 +1,64 @@
+const currencies = [];
+const ratesStorage = {};
+
+const fromSelect = document.getElementById("from-currency");
+const toSelect = document.getElementById("to-currency");
+
+const inputTag = document.getElementById("from-amount");
+const toConverted = document.getElementById("to-amount");
+
+async function fetchCurrencyRates(base = "EUR") {
+    try {
+        const res = await fetch(`https://open.er-api.com/v6/latest/${base}`);
+        const data = await res.json();
+
+        currencies.push(...Object.keys(data.rates));
+
+        return data;
+    } catch (err) {
+        console.error(`Something went wrong: ${err}`);
+    } 
+}
+
+async function generateOptions() {
+    await fetchCurrencyRates();
+    const htmlOptions = currencies
+    .map((option) => {
+        return `<option value="${option}">${option}</option>`;
+    }).join(" ");
+
+    fromSelect.innerHTML = htmlOptions;
+    toSelect.innerHTML = htmlOptions;
+    toSelect.value = "DKK";
+}
+
+generateOptions();
+
+async function convert(amount, from, to) {
+    if(!ratesStorage[from]) {
+        //console.log(`Unable to convert from ${from} to ${to}`);
+        const rates = await fetchCurrencyRates(from);
+        //console.log(rates);
+        ratesStorage[from] = rates;
+    }
+    const rate = ratesStorage[from].rates[to];
```

Comment:
If you are trying to locally cache the rates, which is a nice idea, you can add them to [localStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage) or [sessionStorage](https://developer.mozilla.org/en-US/docs/Web/API/Window/sessionStorage) so that the values persist on refreshing the page too.